commons/all/src/main/java/org/infinispan/commons/configuration/io/AbstractConfigurationWriter.java

```diff
@@ -0,0 +1,70 @@
+package org.infinispan.commons.configuration.io;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.infinispan.commons.util.Util;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 12.1
+ **/
+public abstract class AbstractConfigurationWriter implements ConfigurationWriter {
+   final protected Writer writer;
+   final protected Deque<String> tagStack = new ArrayDeque<>();
+   final protected Map<String, String> namespaces = new LinkedHashMap<>();
```

Comment:
Does this need to be a `LinkedHashMap`? From what I can tell, we never do anything that requires insertion order.